file1.js

```diff
@@ -0,0 +1,181 @@
+/* global QamineUtils */
+
+"use strict";
+
+(function($, window) {
+
+  var utils = QamineUtils.getInstance();
+
+  $(document).ready(function() {
+
+    //Profile
+
+    var bindProfileEdit = function(textClass, buttonClass, feedbackClass) {
+      $(textClass).keyup(function() {
+        $(buttonClass).prop("disabled", false);
+      });
+
+      $(buttonClass).click(function() {
+        var $this = $(this);
+        var postVariable = {};
+        postVariable.name = $(textClass).val();
+
+        var url = $this.data("url");
+        utils.JSend(url, postVariable, function(data) {
+            $(feedbackClass).showSuccess("Saved");
+          }, function(error) {
+            $(feedbackClass).addError(error);
+          },
+          "Could not save. please try again.");
+
+      });
+    };
+    bindProfileEdit(".NameText", ".NameButton", ".NameFeedback");
+    bindProfileEdit(".jsUsernameText", ".jsUsernameButton", ".jsUsernameFeedback");
+    $(".help-tooltip").popover();
+
+    //Vouchers
+    $(".VoucherText").bind("keyup paste", function() {
+      $(".VoucherButton").prop("disabled", false);
+    });
+    $(".VoucherButton").click(function(e) {
+      e.preventDefault();
+
+      var $this = $(this);
+      var postVariable = {};
+      postVariable.code = $(".VoucherText").val();
+
+      var url = $this.data("url");
+      utils.JSend(url, postVariable, function(data) {
+          $(".VoucherFeedback").addSuccess("Promocode added. Your trial was extended.");
+          $(".VoucherText").val("");
+          $(".VoucherButton").prop("disabled", true);
+        }, function(error) {
+          $(".VoucherFeedback").showError(error);
+        },
+        "Could not add the promocode, please try again.");
+    });
+
+    $(".BtnGenerateToken").click(function() {
+      var $this = $(this);
+
+      var url = $this.data("url");
+      utils.JSend(url, {}, function(data) {
```

Comment:
![Codacy](https://app.dev.codacy.org/assets/images/favicon.png) Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor. (new-cap)](https://app.dev.codacy.org/app/codacytestuser/CheckPrCommentsPublicProject/pullRequest?prid=7992)